core/trino-main/src/main/java/io/trino/operator/DeleteAndInsertMergeProcessor.java

```diff
@@ -111,7 +113,7 @@ public Page transformPage(Page inputPage)
                 case DEFAULT_CASE_OPERATION_NUMBER -> { /* ignored */ }
                 case INSERT_OPERATION_NUMBER -> insertPositions++;
                 case DELETE_OPERATION_NUMBER -> deletePositions++;
-                case UPDATE_OPERATION_NUMBER -> updatePositions++;
+                case UPDATE_OPERATION_NUMBER, UPDATE_INSERT_OPERATION_NUMBER, UPDATE_DELETE_OPERATION_NUMBER -> updatePositions++;
```

Comment:
if UPDATE_DELETE_OPERATION_NUMBER is really a delete, shouldn't this increment deletePositions?

Reply:
No it should not, the entire point of this mechanism  is to distinguish between deletes coming from DELETE queries and deletes coming from UPDATE queries. 
Later they are treated in the pretty much same way but generate different CDF output. If CDF is disabled then they are treated in exactly the same way.